src/BizTalk.Extended.Pipelines.General/PipelineComponent.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using BizTalk.Extended.Core.Guards;
+using Microsoft.BizTalk.Component.Interop;
+using Microsoft.BizTalk.Message.Interop;
+
+using IComponent = Microsoft.BizTalk.Component.Interop.IComponent;
+
+namespace BizTalk.Extended.Pipelines.General
+{
+    public abstract class PipelineComponent<T> : IBaseComponent,
+                                                 IPersistPropertyBag,
+                                                 IComponent,
+                                                 IComponentUI
+                                                 where T : PipelineComponent<T>
+    {
+
+        #region Public Properties
+        /// <summary>
+        /// Gets or sets a value to enable (true) or disable (false) the
+        /// pipeline component.
+        /// </summary>
+        [Browsable(true)]
+        [Description("Enables (true) or disables (false) the pipeline component.")]
+        private bool _isEnabled = true;
+        public bool IsEnabled
+        {
+            get { return this._isEnabled; }
+            set { this._isEnabled = value; }
+        }
+        #endregion
+
+        #region Pipeline component
+        /// <summary>
+        /// Method that will contain the actual execute implementation.
+        /// </summary>
+        /// <param name="pielineContext">Context of the pipeline.</param>
+        /// <param name="message">message that passes through the pipeline.</param>
+        /// <returns></returns>
+        protected abstract IBaseMessage ExecuteInternal(IPipelineContext pielineContext, IBaseMessage message);
+
+        /// <summary>
+        /// Public execute metod of the pipeline.
+        /// </summary>
+        /// <param name="pipelineContext"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public IBaseMessage Execute(IPipelineContext pipelineContext, IBaseMessage message)
+        {
+            Guard.NotNull(pipelineContext, "pipelineContext");
+            Guard.NotNull(message, "message");
+
+            IBaseMessage returnMessage = null;
```

Comment:
Maybe it's better to assign it directly to `message` and get rid of the `else` branch in the `try\catch`?